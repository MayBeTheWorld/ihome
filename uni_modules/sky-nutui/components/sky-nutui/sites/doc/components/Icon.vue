<template>
  <div class="icon-moudle">
    <div class="item home" @click="goHome">
      <nut-icon name="home"></nut-icon>
    </div>
    <div class="item refresh" @click="refresh">
      <nut-icon name="refresh"></nut-icon>
    </div>
    <div class="item category" @click="toCategory($event)">
      <nut-icon name="category"></nut-icon>
    </div>
    <div class="qrcodepart" v-if="showCode" ref="codeRef" @click="stopClick($event)">
      <div class="qrcode"> </div>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, reactive, ref, onMounted, toRefs } from 'vue';
import { useRoute, useRouter } from 'vue-router';
export default defineComponent({
  name: 'demo-icon',
  components: {},
  emits: ['refresh', 'goHome'],
  setup(props: any, { emit }: any) {
    const router = useRouter();
    const route = useRoute();

    const state = reactive({
      showCode: false
    });

    const codeRef = ref(null);

    onMounted(() => {
      document.addEventListener('click', (e: any) => {
        state.showCode = false; // 点击其他区域关闭
      });
      // console.log('route', route)
    });

    // 返回demo 首页页
    const goHome = () => {
      // router.push('/');
      emit('goHome');
    };

    // 刷新
    const refresh = () => {
      // router.go(0);
      emit('refresh');
    };

    // 二维码
    const toCategory = (e: any) => {
      state.showCode = !state.showCode;
      e.stopPropagation(); // 阻止事件向上冒泡
    };

    const stopClick = (e: any) => {
      e.stopPropagation(); // 阻止事件向上冒泡
    };

    return reactive({
      ...toRefs(state),
      codeRef,
      goHome,
      refresh,
      toCategory,
      stopClick
    });
  }
});
</script>
<style lang="scss" scoped>
.icon-moudle {
  display: flex;
  justify-content: flex-end;
  position: absolute;
  z-index: 10;
  left: 0;
  right: 0;
  bottom: 0;
  height: 40px;
  padding: 0 22px;
  color: #666;
  border-top: 1px solid #e6e6e6;
  background: #fff;
  .item {
    margin-left: 12px;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
  }
}
.qrcodepart {
  // display: none;
  position: absolute;
  right: 9px;
  bottom: 48px;
  width: 110px;
  padding: 6px 7px;
  border-radius: 6px;
  background: rgba(255, 255, 255, 1);
  box-shadow: 0px 4px 10px 0px rgba(0, 0, 0, 0.11);
  box-sizing: border-box;
  .qrcode {
    width: 96px;
    height: 96px;
    margin: 0 auto;
    background: url(https://img12.360buyimg.com/imagetools/jfs/t1/162421/39/13392/9425/6052ea60E592310a9/264bdff23ef5fe95.png)
      no-repeat;
    background-size: cover;
  }
}
</style>
