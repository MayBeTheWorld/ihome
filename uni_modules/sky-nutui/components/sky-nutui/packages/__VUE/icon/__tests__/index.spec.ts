import { config, mount } from '@vue/test-utils';
import Icon from '../index.vue';

test('should render icon dongdong', async () => {
  const wrapper = mount(Icon, {
    props: {
      name: 'dongdong'
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});
test('should render icon image type', async () => {
  const wrapper = mount(Icon, {
    props: {
      name: 'https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png'
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render icon dongdong color', async () => {
  const wrapper = mount(Icon, {
    props: {
      name: 'dongdong',
      color: 'red'
    }
  });

  expect(wrapper.find<HTMLElement>('.nut-icon-dongdong').element.style.color).toEqual('red');
  expect(wrapper.html()).toMatchSnapshot();
});

test('should render icon dongdong size', async () => {
  const wrapper = mount(Icon, {
    props: {
      name: 'dongdong',
      size: '24'
    }
  });

  expect(wrapper.find<HTMLElement>('.nut-icon-dongdong').element.style.fontSize).toEqual('24px');
  expect(wrapper.html()).toMatchSnapshot();
});
