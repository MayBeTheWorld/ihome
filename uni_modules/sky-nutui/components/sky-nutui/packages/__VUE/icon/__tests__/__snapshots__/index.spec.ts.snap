// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render icon dongdong 1`] = `"<i class=\\"nutui-iconfont nut-icon nut-icon-dongdong\\" src=\\"\\"></i>"`;

exports[`should render icon dongdong color 1`] = `"<i class=\\"nutui-iconfont nut-icon nut-icon-dongdong\\" style=\\"color: red;\\" src=\\"\\"></i>"`;

exports[`should render icon dongdong size 1`] = `"<i class=\\"nutui-iconfont nut-icon nut-icon-dongdong\\" style=\\"font-size: 24px; width: 24px; height: 24px;\\" src=\\"\\"></i>"`;

exports[`should render icon image type 1`] = `"<img class=\\"nut-icon__img\\" src=\\"https://img11.360buyimg.com/imagetools/jfs/t1/137646/13/7132/1648/5f4c748bE43da8ddd/a3f06d51dcae7b60.png\\">"`;
